### ISSUE TITLE
Looping over unbounding array may lead to DoS

### ISSUE THREAD

**Issue by c4-bot-1** [code-423n4/2024-05-munchables-validation#177]
# Lines of code

https://github.com/code-423n4/2024-05-munchables/blob/57dff486c3cd905f21b330c2157fe23da2a4807d/src/managers/LockManager.sol#L115-L122

# Vulnerability details

## Impact

According to provided documentation, protocol does not implement any trusted role, which implicitly implies, that every role within the protocol should be treated as a non-trusted:

[File: README.md](https://github.com/code-423n4/2024-05-munchables/blob/main/README.md#all-trusted-roles-in-the-protocol)
```
All trusted roles in the protocol

N/A
```

A non-trusted `onlyAdmin` role can call `configureToken()` multiple of times - adding many tokens to `configuredTokenContracts` array.

Protocol implements a loop on `configuredTokenContracts` array in many places. When `configuredTokenContracts` array is too big, protocol will revert with out of gas error - thus noone will be able to use properly use the smart contract.

## Proof of Concept

1. Non trusted admin adds many tokens to `configuredTokenContracts` by calling `configureToken()` function multiple of times:

[File: LockManager.sol](https://github.com/code-423n4/2024-05-munchables/blob/57dff486c3cd905f21b330c2157fe23da2a4807d/src/managers/LockManager.sol#L115-L122)
```
function configureToken(
        address _tokenContract,
        ConfiguredToken memory _tokenData
    ) external onlyAdmin {
        if (_tokenData.nftCost == 0) revert NFTCostInvalidError();
        if (configuredTokens[_tokenContract].nftCost == 0) {
            // new token
            configuredTokenContracts.push(_tokenContract);
```

Looping over unbounded array (when that array is big) will result in out of gas error. The genuine user won't be able to call:

* `setLockDuration()`

[File: LockManager.sol](https://github.com/code-423n4/2024-05-munchables/blob/57dff486c3cd905f21b330c2157fe23da2a4807d/src/managers/LockManager.sol#L251-L252)
```
        uint256 configuredTokensLength = configuredTokenContracts.length;
        for (uint256 i; i < configuredTokensLength; i++) {
```

Calling `setLockDuration()` will revert with out of gas error, thus it won't be possible to set lock durations.

* `getLocked()`, `getLockedWeightedValue()`

[File: LockManager.sol](https://github.com/code-423n4/2024-05-munchables/blob/57dff486c3cd905f21b330c2157fe23da2a4807d/src/managers/LockManager.sol#L433-L438)
```
        uint256 configuredTokensLength = configuredTokenContracts.length;
        LockedTokenWithMetadata[]
            memory tmpLockedTokens = new LockedTokenWithMetadata[](
                configuredTokensLength
            );
        for (uint256 i; i < configuredTokensLength; i++) {
```

[File: LockManager.sol](https://github.com/code-423n4/2024-05-munchables/blob/57dff486c3cd905f21b330c2157fe23da2a4807d/src/managers/LockManager.sol#L465-L467)
```
        uint256 configuredTokensLength = configuredTokenContracts.length;
        for (uint256 i; i < configuredTokensLength; i++) {
```

Calling `getLocked()` and `getLockedWeightedValue()` will revert with out of gas error, thus it won't be possible to list locked tokens for each player and locked weight value for each player.

In other words - contract will be bricked - it won't be possible to use the contract at all, since the main functions from the contract will revert with out of gas error.

## Tools Used

Manual code review

## Recommended Mitigation Steps

Limit the number of tokens which can be added to `configuredTokenContracts`

## Assessed type

DoS
